pkg/registries/docker/docker.go

```diff
@@ -22,8 +22,22 @@ import (
 )
 
 const (
-	registryTimeout  = 5 * time.Second
-	repoListInterval = 10 * time.Minute
+	// registryDialerTimeout is the net.Dialer timeout of the client transport.
+	// It limits the time the dialer attempts the connection. The timeout is
+	// chosen as a small value to prevent unavailable registries from blocking
+	// image scanning.
+	registryDialerTimeout = 5 * time.Second
```

Comment:
AFAIU in case we use proxies, the dialer timeout would be for the proxy URL.

What we've seen in some cases was that dialing the proxy URL worked fine, however reaching the proxied service may be unresponsive (depending on the way the proxy is configured) - would that fall now in the dial timeout, or the registry response timeout?

Reply:
I don't know, let me try to test that.

Reply:
I set up a non-existing registry:
* Without squid proxy we are hitting the dial timeout as expected.
* With squid proxy (allow all) we are either hitting squid proxy's `connection_timeout` or Central's client timeout. Strangely enough we are not hitting the response timeout in this case.